src/components/Link/Link.scss

```diff
@@ -49,3 +49,10 @@
 .ms-Link {
   @include ms-Link;
 }
+
+// For button links we need to ensure by default that we inherit the font size. Otherwise
+// the button default font size will override.
+
+BUTTON.ms-Link {
+  font-size: inherit;
```

Comment:
With the change, it looks correct:

![image](https://cloud.githubusercontent.com/assets/1110944/18758235/0bac78d8-80ac-11e6-8baf-11aedcd7c560.png)